### PR TITLE
[FE]: Dashboard Page UI has been created

### PR DESCRIPTION
**Pull request by drakkomaximo**
### StarShop Dashboard Progress

**Implemented Features:**

1. **Dashboard Layout:**  
   - The main structure for the dashboard is set up using a clean card-based design to display key sections like store information, top products, and sales statistics.

2. **About Your Store:**  
   - A section showing basic details about the store, including the store's name and a brief description.

3. **Top Products:**  
   - A section to display the store's featured or top-selling products.

4. **Store Statistics:**  
   - A section dedicated to key metrics showing the store's performance, like sales data.

5. **Monthly Sales:**  
   - A section to show the store's sales performance for each month.

6. **Responsive Design:**  
   - The layout adjusts well to different screen sizes, from mobile to desktop, ensuring a good user experience on all devices.

7. **Reusable Components:**  
   - Using a component-based design to keep the dashboard flexible and easy to update.

8. **Centered Content:**  
   - Ensured the content is always centered and doesn’t stretch beyond a maximum width of 1440px for consistency.

**What to Verify:**
- Check how the dashboard looks on different screen sizes.
- Make sure images load correctly.
- Test all links and buttons to ensure they work properly.
- Ensure text is easy to read and has good contrast.

image
image

### CODE REVIEW

File: src/components/dashboard/DashboardComponent.tsx [hunk 1]
```diff
@@ -1,10 +1,73 @@
...
-        <>
-            <h1 className="text-white text-2xl font-bold">Dashboard</h1>
-        </>
-    )
-}
-
-export default DashboardComponent
\ No newline at end of file
+  return (
+    <div className="min-h-screen space-y-24 bg-[#1a1a2e] text-white p-4 sm:p-6 lg:p-8">
```
Comment (KevinLatino) [inline src/components/dashboard/DashboardComponent.tsx:13]:
```suggestion
    <div className="min-h-screen space-y-24 text-white p-4 sm:p-6 lg:p-8">
```
please, delete the bg-color

File: src/components/dashboard/components/CardContainer.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+// Attach SubTitle to Body
+Body.SubTitle = SubTitle;
+
+const Footer: FC<CardContainerSectionProps> = ({ children }) => {
+  return <div className="mt-4">{children}</div>;
+};
+
+// Attach subcomponents to CardContainer
+CardContainer.Header = Header;
```
Comment (KevinLatino) [inline src/components/dashboard/components/CardContainer.tsx:81]: Please, delete subcomponents, and all it's implementations

**Comment by drakkomaximo** [comment]
# Summary of Changes

## 1. **CardContainer Adjustments**
   - Refactored `CardContainer` from a functional wrapper to a normal component.
   - Handled sections like "Sells Per Month" differently in the component hierarchy.

## 2. **Dashboard Component Layout**
   - Updated the layout of `DashboardComponent` to use the new structure of `CardContainer`.
   - Adjusted child components such as `TopProducts`, `AboutStore`, `Statistics`, and `SellsPerMonth`.
   - Reorganized sections in the `DashboardComponent` to ensure a responsive grid layout.

## 3. **Chart Update - SellsPerMonth**
   - Changed the chart type from a line chart to a donut chart (circular bar chart).
   - Displayed five specific metrics: "Trustless Work", "SotNet", "Safetrust", "ScanGuard", and "Other".
   - Selected custom colors for each section of the donut chart.

These updates improve the component structure, UI elements, and responsiveness across the application.

File: src/components/dashboard/components/CardContainer.tsx (hunk 2 above)
Comment (drakkomaximo) [reply src/components/dashboard/components/CardContainer.tsx]: Done

File: src/components/dashboard/DashboardComponent.tsx (hunk 1 above)
Comment (drakkomaximo) [reply src/components/dashboard/DashboardComponent.tsx]: Done

File: src/components/dashboard/components/CardContainer.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+          <h4 className="text-center text-lg sm:text-xl lg:text-2xl font-semibold mt-2">
+            {subtitle}
+          </h4>
+        )}
+        {children}
+      </div>
+
+      {/* Footer Section */}
+      {footer && <div className="mt-4">{footer}</div>}
+    </div>
```
Comment (KevinLatino) [inline src/components/dashboard/components/CardContainer.tsx:53]:
```suggestion
    <div className={`flex flex-col justify-end ${className}`}>
      {/* Header Section */}
      {(icon || title) && (
        <div
          className={`flex gap-2 items-center mb-2 pl-4 ${
            !icon && !title ? "h-[44px]" : ""
          }`}
        >
          {icon && <div>{icon}</div>}
          {title && (
            <h3 className="text-[20px] sm:text-[24px] lg:text-[30px] font-bold text-center text-background">
              {title}
            </h3>
          )}
        </div>
      )}

      {/* Body Section */}
      <div className="flex flex-col items-center px-4 pb-4 border-4 border-custom-light-card-border rounded-[20px] sm:rounded-[24px] lg:rounded-[28px] max-h-[254px] h-screen">
        {subtitle && (
          <h4 className="text-center text-lg sm:text-xl lg:text-2xl font-semibold mt-2">
            {subtitle}
          </h4>
        )}
        {children}
      </div>

      {/* Footer Section */}
      {footer && <div className="mt-4">{footer}</div>}
    </div>
```
Make this more flexible, don't use an estructure (title, icon, footer) just use children prop

**Review by KevinLatino — CHANGES_REQUESTED** [review]
Please delete all the comments

File: src/components/dashboard/components/CardContainer.tsx (hunk 3 above)
Comment (drakkomaximo) [reply src/components/dashboard/components/CardContainer.tsx]: The behavior of the CardContainer was adjusted to make it a simple container.

**Comment by drakkomaximo** [comment]
image

**Pull request merged**